mailpoet/tests/DataFactories/WooCommerceOrder.php

```diff
@@ -95,16 +95,37 @@ public function create() {
     $orderOut = $this->tester->cliToString(['wc', 'shop_order', 'get', $createOutput, '--format=json', '--user=admin']);
     $order = json_decode($orderOut, true);
     if (isset($this->data['date_created']) && is_array($order) && is_int($order['id'])) {
-      $wcOrder = wc_get_order($order['id']);
-      if ($wcOrder instanceof \WC_Order) {
-        $wcOrder->set_date_created(get_gmt_from_date($this->data['date_created']));
-        $wcOrder->set_date_modified(get_gmt_from_date($this->data['date_created']));
-        $wcOrder->save();
+      if (function_exists('wc_get_order')) {
+        $wcOrder = wc_get_order($order['id']);
+        if ($wcOrder instanceof \WC_Order) {
+          $wcOrder->set_date_created(get_gmt_from_date($this->data['date_created']));
+          $wcOrder->set_date_modified(get_gmt_from_date($this->data['date_created']));
+          $wcOrder->save();
+        }
+      } else {
+        $this->updateOrderCreateDateWhenWooIsNotAvailable($order['id'], $this->data['date_created']);
       }
     }
     return $order;
   }
 
+  /**
+   * This is a workaround to update an order even when WooCommerce is not active. This was necessary
+   * for the analytics cest in MailPoet premium. This workaround does not work in a COT environment.
+   *
+   * @param int $orderId
+   * @param string $date
+   * @return void
+   */
+  private function updateOrderCreateDateWhenWooIsNotAvailable(int $orderId, string $date) {
+    $cmd = ['post', 'update', '--porcelain', $orderId ];
+    $cmd[] = '--post_date="' . $date . '"';
+    $cmd[] = '--post_date_gmt="' . $date . '"';
+    $cmd[] = '--post_modified="' . $date . '"';
+    $cmd[] = '--post_modified_gmt="' . $date . '"';
+    $this->tester->cliToString($cmd);
+  }
+
```

Comment:
Why would we need to update an order without Woo? 🤔

Reply:
Ahh, fixed in https://github.com/mailpoet/mailpoet-premium/pull/752/commits/a7878e83fba12a93a778b4d4e7d86b548e945124.